### ISSUE TITLE
3D is inverted

### ISSUE THREAD

**Issue by DerpedCrusader** [elhobbs/prboom3ds#2]
No idea how to show this or anything, but on my Old 3DS Xl, the 3D in the game is inverted, and a fix for that would be nice

**Comment by elhobbs** [comment]
Make sure you are using the latest version. The initial release the 3d slider did not work correctly. The slider all the way up was inverted, middle was neutral, and down was correct 3d. 

The current version has issues with the view sprites.

**Comment by DerpedCrusader** [comment]
is there another place to get that version, because i just got it for the first time last week,

**Comment by elhobbs** [comment]
I have only put files on github. The n3ds branch is the newest version (I am guessing that the newest version is what you mean by "that" version).

**Comment by DerpedCrusader** [comment]
Nope, i was using the master, thanks for pointing out the branches thing, i'm new to github and yeah, if you hadnt said that, i would just been like "well, he's crazy" and went on my way, thanks man

**Comment by LAGonauta** [comment]
Hello, just tried v0.2-alpha.3dsx on a O3DS on 11.0.0 with A9LH and the 3D view seems to still have this bug, and I couldn't find the n3ds branch (just a n3ds fork with no commit fixing that).
This is the only commit that I found about the 3D: https://github.com/elhobbs/prboom3ds/commit/796098b54ec55c530a1a52e914a7053ba9d2ec49

Was this ever fixed but the fix does not work on the O3DS? The rendering seems to be working fine when I tilt a little my 3DS to physically invert the image.

**Comment by elhobbs** [comment]
I never fixed the 3D effect for the view model or the menus - so they look a little off. 

The world looks fine to me though. Maybe I am not understanding the issue you are referring to though. I am not sure how tilting the 3ds would fix anything. The o3ds in particular only works correctly when viewed at very precise angles - you have to be looking straight on. 

If I remember correctly the commit you referenced caused the left and right frames to alternate repeatedly.

**Comment by LAGonauta** [comment]
At first I thought that the images for the left and right eyes were swapped on my unit for some reason, so something that is far away seemed to be close and something that is near seem to be far away. By tilting the O3DS a little I can force-swap the images so everything would be seen at the correct depth.

But testing now there seems to actually be some kind of rendering bug.
English is not my native language so I will try to explain it using some numbers.
Let's say that depth can go from -15 cm (nose of the player) to 15 (15 cm into the screen). Zero is at the screen.

I compared both rendered images and stuff far away are being rendered at zero (at the screen) and anything that should be closer to the player are being rendered outside of the screen, close to the player nose creating a nasty effect.
I uploaded some pictured to this post.

Basically, everything that is far away is being rendered at the depth of the menu and the weapons, and everything that is closer is being rendered close to the nose of the player.

Left eye:
![scr_1_top_left](https://cloud.githubusercontent.com/assets/14236067/16891013/a5a7b00a-4aca-11e6-855c-2e6a106ddcad.png)

Right eye:
![scr_1_top_right](https://cloud.githubusercontent.com/assets/14236067/16891014/a5d00046-4aca-11e6-8198-2ec411120075.png)

 nose.

**Comment by elhobbs** [comment]
The left frame is generated by shifting the view position to the left and the right frame is produced by shifting the view position to the right. This does not work for items that are close to the view point and it really messes up the sprites when they are close to the view point as they can be showing different images. And as I mentioned previously the menu and viewsprite( the gun) look all wrong. If you ignore the gun and look at the center of the screen the 3D effect is passable. You do have to play with the slider a bit as the full setting as way too severe. 

I agree that this is a bad approach. A better approach would require considerable modifications to the rendering engine. 

Feel free to fix it :)

**Comment by LAGonauta** [comment]
Yeah, I guess the pseudo-3D engine from Doom isn't accurate enough for stereoscopic rendering. Just like trying to look up and down on Star Wars: Dark Forces.. Oh my, that gives me headaches.
